core/src/test/java/org/elasticsearch/search/functionscore/FunctionScoreIT.java

```diff
@@ -17,61 +17,94 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
 import static org.elasticsearch.client.Requests.searchRequest;
 import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
 import static org.elasticsearch.index.query.QueryBuilders.functionScoreQuery;
 import static org.elasticsearch.index.query.QueryBuilders.termQuery;
 import static org.elasticsearch.index.query.functionscore.ScoreFunctionBuilders.scriptFunction;
+import static org.elasticsearch.script.ScriptService.ScriptType;
 import static org.elasticsearch.search.aggregations.AggregationBuilders.terms;
 import static org.elasticsearch.search.builder.SearchSourceBuilder.searchSource;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchResponse;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 
-public class FunctionScoreTests extends ESIntegTestCase {
+public class FunctionScoreIT extends ESIntegTestCase {
+
     static final String TYPE = "type";
     static final String INDEX = "index";
 
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singleton(GroovyPlugin.class);
+        return Collections.singleton(CustomScriptPlugin.class);
+    }
+
+    public static class CustomScriptPlugin extends MockScriptPlugin {
+
+        @Override
+        @SuppressWarnings("unchecked")
+        protected Map<String, Function<Map<String, Object>, Object>> pluginScripts() {
+            Map<String, Function<Map<String, Object>, Object>> scripts = new HashMap<>();
+            scripts.put("1", vars -> 1.0d);
+            scripts.put("_score.doubleValue()", vars -> ((ScoreAccessor) vars.get("_score")).doubleValue());
+            scripts.put("return (doc['num'].value)", vars -> {
+                Map<?, ?> doc = (Map) vars.get("doc");
+                ScriptDocValues.Longs num = (ScriptDocValues.Longs) doc.get("num");
+                return num.getValue();
+            });
+            scripts.put("doc['random_score']", vars -> {
+                Map<?, ?> doc = (Map) vars.get("doc");
+                ScriptDocValues.Doubles randomScore = (ScriptDocValues.Doubles) doc.get("random_score");
+                return randomScore.getValue();
+            });
+            return scripts;
+        }
     }
 
     public void testScriptScoresNested() throws IOException {
         createIndex(INDEX);
         ensureYellow();
         index(INDEX, TYPE, "1", jsonBuilder().startObject().field("dummy_field", 1).endObject());
         refresh();
+
+        Script scriptOne = new Script("1", ScriptType.INLINE, CustomScriptPlugin.NAME, null);
+        Script scriptTwo = new Script("_score.doubleValue()", ScriptType.INLINE, CustomScriptPlugin.NAME, null);
```

Comment:
Since the id can be any string, I prefer to rename to rename to an explicit definition like "get score value" :)